src/features/codeActionProvider.ts

```diff
@@ -101,80 +100,12 @@ export default class CodeActionProvider extends AbstractProvider implements vsco
 
     private async _runCodeAction(req: protocol.V2.RunCodeActionRequest, token: vscode.CancellationToken): Promise<boolean | string | {}> {
 
-        return serverUtils.runCodeAction(this._server, req).then(async response => {
-            if (response && Array.isArray(response.Changes)) {
-
-                let edit = new vscode.WorkspaceEdit();
-
-                let fileToOpen: Uri = null;
-                let renamedFiles: Uri[] = [];
-
-                for (let change of response.Changes) {
-                    if (change.ModificationType == FileModificationType.Renamed)
-                    {
-                        // The file was renamed. Omnisharp has already persisted
-                        // the right changes to disk. We don't need to try to
-                        // apply text changes (and will skip this file if we see an edit)
-                        renamedFiles.push(Uri.file(change.FileName));
-                    }
-                }
-
-                for (let change of response.Changes) {
-                    if (change.ModificationType == FileModificationType.Opened)
-                    {
-                        // The CodeAction requested that we open a file.
-                        // Record that file name and keep processing CodeActions.
-                        // If a CodeAction requests that we open multiple files
-                        // we only open the last one (what would it mean to open multiple files?)
-                        fileToOpen = vscode.Uri.file(change.FileName);
-                    }
-
-                    if (change.ModificationType == FileModificationType.Modified)
-                    {
-                        let uri = vscode.Uri.file(change.FileName);
-                        if (renamedFiles.some(r => r == uri))
-                        {
-                            // This file got renamed. OmniSharp has already
-                            // persisted the new file with any applicable changes.
-                            continue;
-                        }
-
-                        let edits: vscode.TextEdit[] = [];
-                        for (let textChange of change.Changes) {
```

Comment:
This logic is unchanged, other than using `const` where appropriate and correcting for the necessary cast on `change` to the correct type.